### PR TITLE
[vividus] Register listeners with subscribers in super classses

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus/src/main/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessor.java [hunk 1]
```diff
@@ -29,13 +29,16 @@
...
+    @SuppressWarnings("unchecked")
     @Override
     public Object postProcessAfterInitialization(Object bean, String beanName)
     {
-        if (Stream.of(bean.getClass().getDeclaredMethods()).anyMatch(m -> m.isAnnotationPresent(Subscribe.class)))
-        {
-            eventBus.register(bean);
-        }
+        Stream.iterate((Class<Object>) bean.getClass(), c -> c.getSuperclass() != null, Class::getSuperclass)
+              .map(Class::getDeclaredMethods)
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessor.java:37]: what about just `getMethods`?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus/src/main/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessor.java (hunk 1 above)
Comment (uarlouski) [reply vividus/src/main/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessor.java]: sure, I don't know why it was getDeclaredMethods previously, but from docs event bus works with public handler methods

File: vividus/src/main/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessor.java [hunk 2]
```diff
@@ -18,24 +18,30 @@
...
 
+    @SuppressWarnings("unchecked")
     @Override
     public Object postProcessAfterInitialization(Object bean, String beanName)
     {
-        if (Stream.of(bean.getClass().getDeclaredMethods()).anyMatch(m -> m.isAnnotationPresent(Subscribe.class)))
-        {
-            eventBus.register(bean);
-        }
+        Stream.iterate((Class<Object>) bean.getClass(), c -> c.getSuperclass() != null, Class::getSuperclass)
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessor.java:39]: if you use `getMethods`, you don't need to extract superclass explicitly

File: vividus/src/test/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessorTests.java [hunk 3]
```diff
@@ -37,12 +40,13 @@
...
-    @Test
-    void testProcessingSubscriberObject()
+    @ValueSource(classes = { Subscriber.class, InheritedSubscriber.class })
+    @ParameterizedTest
+    void testProcessingSubscriberObject(Class<?> type)
     {
-        Subscriber subscriber = new Subscriber();
-        beanPostProcessor.postProcessAfterInitialization(subscriber, null);
-        verify(eventBus).register(subscriber);
+        Object bean = ReflectionUtils.newInstance(type);
```
Comment (valfirst) [inline vividus/src/test/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessorTests.java:47]: just objects as test arguments

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**